src/login/logind-dbus.c

```diff
@@ -983,6 +997,49 @@ static int method_create_session(sd_bus_message *message, void *userdata, sd_bus
         return r;
 }
 
+
+
+static int method_create_session(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        const char *service, *type, *class, *cseat, *tty, *display, *remote_user, *remote_host, *desktop;
+        pid_t leader;
+        uid_t uid;
+        int remote;
+        uint32_t vtnr = 0;
+        int r;
+
+        assert(message);
+
+        assert_cc(sizeof(pid_t) == sizeof(uint32_t));
+        assert_cc(sizeof(uid_t) == sizeof(uint32_t));
+
+        r = sd_bus_message_read(message, "uusssssussbss",
+                                &uid, &leader, &service, &type, &class, &desktop, &cseat,
+                                &vtnr, &tty, &display, &remote, &remote_user, &remote_host);
+        if (r < 0)
+                return r;
+
+        return create_session(message, userdata, error,
+                              uid, leader, -EBADF, service, type, class, desktop, cseat, vtnr, tty, display, remote, remote_user, remote_host);
+}
+
+static int method_create_session_pidfd(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        const char *service, *type, *class, *cseat, *tty, *display, *remote_user, *remote_host, *desktop;
+        int leaderfd = -EBADF;
+        uid_t uid;
+        int remote;
+        uint32_t vtnr = 0;
+        int r;
+
+        r = sd_bus_message_read(message, "uhsssssussbss",
+                                &uid, &leaderfd, &service, &type, &class, &desktop, &cseat,
+                                &vtnr, &tty, &display, &remote, &remote_user, &remote_host);
+        if (r < 0)
+                return r;
+
+        return create_session(message, userdata, error,
+                              uid, -1, leaderfd, service, type, class, desktop, cseat, vtnr, tty, display, remote, remote_user, remote_host);
```

Comment:
we usually use 0 as invalid pid. should really do here too. given you specify the leaderfd  here that should be fine

Reply:
`/* pid= */ 0`